### ISSUE TITLE
feat: Upgrade the generator

### ISSUE THREAD

**Issue by Njord201** [FppEpitech/MyTorch#27]
FEAT: Upgrade the generator.

- [ ] Add the learning rate of the neural network.
- [ ] Add for each neuron, the activation function.
- [ ] number of outputs neurons
